### PR TITLE
Added states and config for re-deploying starcellbio

### PR DESCRIPTION
**Pull request by blarghmatey**
#### What are the relevant tickets?
#682 

#### What's this PR do?
Adds states and configs to deploy starcellbio in rc/prod apps environments

### CODE REVIEW

File: pillar/apps/starcellbio.sls [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    DB_ENGINE: django.db.backends.mysql
+    DB_NAME: starcellbio
+    DB_USER: __vault__:cache:mariadb-{{ ENVIRONMENT }}-starcellbio/creds/starcellbio>data>username
+    DB_PASSWORD: __vault__:cache:mariadb-{{ ENVIRONMENT }}-starcellbio/creds/starcellbio>data>password
+    DB_HOST: mariadb-starcellbio.service.consul
+    DB_PORT: 3306
+    TEMPLATE_DEBUG: false
+    SERVER_EMAIL: mitxmail@mit.edu
+    ADMINS:
+      -
```
Comment (shaidar) [inline pillar/apps/starcellbio.sls]: this looks odd

File: pillar/apps/starcellbio.sls [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        - mitx-devops
+        - mitx-devops@mit.edu
```
Comment (shaidar) [inline pillar/apps/starcellbio.sls]: Why one with email and one without?

File: pillar/nginx/starcellbio.sls [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+{% set env_settings = salt.cp.get_file_str("salt://environment_settings.yml")|load_yaml %}
+{% set ENVIRONMENT = salt.grains.get('environment', 'rc-apps') %}
+{% set env_data = env_settings.environments[ENVIRONMENT] %}
+{% set server_domain_names = env_data.purposes['odl-video-service'].domains %}
+{% set ovs_login_path = 'login' %}
+
+nginx:
+  ng:
+    install_from_source: True
+    source_version: 1.13.8
```
Comment (shaidar) [inline pillar/nginx/starcellbio.sls]: Why not a newer version?

File: pillar/nginx/starcellbio.sls [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                - root: /opt/{{ app_name }}/
+                - ssl_certificate: /etc/nginx/ssl/odl_wildcard.crt
+                - ssl_certificate_key: /etc/nginx/ssl/odl_wildcard.key
+                - ssl_stapling: 'on'
+                - ssl_stapling_verify: 'on'
+                - ssl_session_timeout: 1d
+                - ssl_session_tickets: 'off'
+                - ssl_protocols:
+                    - TLSv1
+                    - TLSv1.1
```
Comment (shaidar) [inline pillar/nginx/starcellbio.sls]: Should we remove both of these?

File: pillar/nginx/starcellbio.sls [hunk 5]
```diff
@@ -0,0 +1,74 @@
+{% set app_name = 'starcellbio' %}
+{% set env_settings = salt.cp.get_file_str("salt://environment_settings.yml")|load_yaml %}
+{% set ENVIRONMENT = salt.grains.get('environment', 'rc-apps') %}
+{% set env_data = env_settings.environments[ENVIRONMENT] %}
+{% set server_domain_names = env_data.purposes['odl-video-service'].domains %}
```
Comment (shaidar) [inline pillar/nginx/starcellbio.sls]: Copy-pasta?

File: salt/environment_settings.yml [hunk 6]
```diff
@@ -424,6 +424,12 @@ environments:
           - elasticsearch-production-apps.odl.mit.edu
         security_groups:
           - webapp
+      starcellbio:
+        business_unit: starteam
+        domains:
+          - starcellbio.mit.edu
```
Comment (shaidar) [inline salt/environment_settings.yml]: Wouldn't using the odl wildcard cert cause mismatch problems for this domain?

File: pillar/apps/starcellbio.sls (hunk 1 above)
Comment (blarghmatey) [reply pillar/apps/starcellbio.sls]: It's the syntax for creating a list of lists in YAML :man_shrugging:

File: pillar/apps/starcellbio.sls (hunk 2 above)
Comment (blarghmatey) [reply pillar/apps/starcellbio.sls]: It's two elements of a tuple, name and email

File: pillar/nginx/starcellbio.sls (hunk 3 above)
Comment (blarghmatey) [reply pillar/nginx/starcellbio.sls]: :spaghetti:

File: pillar/nginx/starcellbio.sls (hunk 4 above)
Comment (blarghmatey) [reply pillar/nginx/starcellbio.sls]: I suppose we can give it a try, but I anticipate that leading to a support request...

File: pillar/nginx/starcellbio.sls (hunk 5 above)
Comment (blarghmatey) [reply pillar/nginx/starcellbio.sls]: :spaghetti:

File: salt/environment_settings.yml (hunk 6 above)
Comment (blarghmatey) [reply salt/environment_settings.yml]: Good point...

File: pillar/nginx/odlvideo.sls [hunk 7]
```diff
@@ -69,7 +69,7 @@ nginx:
                      :DHE-RSA-AES256-SHA:ECDHE-ECDSA-DES-CBC3-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES128-GCM-SHA256\
                      :AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:DES-CBC3-SHA:!DSS"
                 - ssl_prefer_server_ciphers: 'on'
-                - resolver: 8.8.8.8
+                - resolver: 1.1.1.1
```
Comment (shaidar) [inline pillar/nginx/odlvideo.sls]: Is this file supposed to be in this PR?

**Comment by shaidar** [comment]
👍

**Pull request merged**